### ISSUE TITLE
[Bug]: Api with mask not working good

### ISSUE THREAD

**Issue by pakresi** [Mikubill/sd-webui-controlnet#1640]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Hi

```python
payload = {
  "init_images": [
    inputImageBase64
  ],
  "resize_mode": 0,
  "denoising_strength": 1,
  "image_cfg_scale": 7,
  "mask": maskImageBase64,
  "mask_blur": 0,
  "inpainting_fill": 3,
  "inpaint_full_res": True,
  "inpaint_full_res_padding": 0,
  "inpainting_mask_invert": 0,
  "initial_noise_multiplier": 0,
  "prompt": prompt,
  "seed": -1,
  "subseed": -1,
  "subseed_strength": 0,
  #"seed_resize_from_h": int(orig_w),
  #"seed_resize_from_w": int(orig_h),
  "sampler_name": "DPM++ 2M Karras",
  "batch_size": 1,
  "n_iter": 1,
  "steps": steps,
  "cfg_scale": 7,
  "width": int(orig_w),
  "height": int(orig_h),
  "restore_faces": False,
  "tiling": False,
  "do_not_save_samples": False,
  "do_not_save_grid": False,
  "negative_prompt": "string",
  "eta": 0,
  "s_min_uncond": 0,
  "s_churn": 0,
  "s_tmax": 0,
  "s_tmin": 0,
  "s_noise": 1,
  "override_settings": {},
  "override_settings_restore_afterwards": True,
  "script_args": [],
  "sampler_index": "DPM++ 2M Karras",
  "include_init_images": True,
#  "script_name": "string",
  "send_images": True,
  "save_images": False,
    
  "alwayson_scripts": {
               "controlnet": {
                   "args": [
                       {
                        "input_image": inputImageBase64,
                        "module": "canny",
                        "model": "control_v11p_sd15_canny [d14c016b]",
                        "weight": 0.25,
                        "lowvram": False,
                        "threshold_a": 100,
                        "threshold_b": 200,
                        "guidance_start": 0,
                        "guidance_end": 0.6,
                        "control_mode": "My prompt is more important",
                        "resize_mode": 0,
                         "pixel_perfect" : False
                      }
                    ]
                }
            }
    
  }
``` 

when i use mask in main payload, controlnet scopes masked area and wrong crop, i tried all resize_modes result same.

### with mask: (not like webui result with same mask)
<img src="https://github.com/Mikubill/sd-webui-controlnet/assets/133514291/5595a756-3341-4cb9-b17a-1e22ae678aff" width=30% height=30%>

### without mask: (same result with webui)
<img src="https://github.com/Mikubill/sd-webui-controlnet/assets/133514291/2c0930e8-6149-466d-9ddc-acfd5a52ca9c" width=30% height=30%>

### Steps to reproduce the problem

1. try my code

### What should have happened?

control image must be full like without masked version

### Commit where the problem happens

webui: v.1.3.2
controlnet: v1.1.219

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--listen --enable-insecure-extension-access --xformers --opt-split-attention --api
```

### List of enabled extensions

standart

### Console logs

```Shell
without mask
2023-06-14 13:42:10,968 - ControlNet - INFO - Loading model: control_v11p_sd15_canny [d14c016b]
2023-06-14 13:42:17,105 - ControlNet - INFO - Loaded state_dict from [/home/banias/stable-diffusion-webui/models/ControlNet/control_v11p_sd15_canny.pth]
2023-06-14 13:42:17,110 - ControlNet - INFO - Loading config: /home/banias/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_canny.yaml
2023-06-14 13:42:19,791 - ControlNet - INFO - ControlNet model control_v11p_sd15_canny [d14c016b] loaded.
2023-06-14 13:42:19,819 - ControlNet - INFO - API used default config: unit.processor_res = 512
2023-06-14 13:42:19,820 - ControlNet - INFO - Loading preprocessor: canny
2023-06-14 13:42:19,820 - ControlNet - INFO - preprocessor resolution = 512



with mask 
2023-06-14 13:39:17,024 - ControlNet - INFO - Loading model: control_v11p_sd15_canny [d14c016b]
2023-06-14 13:39:17,394 - ControlNet - INFO - Loaded state_dict from [/home/banias/stable-diffusion-webui/models/ControlNet/control_v11p_sd15_canny.pth]
2023-06-14 13:39:17,395 - ControlNet - INFO - Loading config: /home/banias/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_canny.yaml
2023-06-14 13:39:20,106 - ControlNet - INFO - ControlNet model control_v11p_sd15_canny [d14c016b] loaded.
2023-06-14 13:39:20,178 - ControlNet - INFO - API used default config: unit.processor_res = 512
2023-06-14 13:39:20,179 - ControlNet - INFO - Loading preprocessor: canny
2023-06-14 13:39:20,179 - ControlNet - INFO - preprocessor resolution = 512
```

### Additional information

How can i fix it?

**Comment by sclausen** [comment]
I would be curious about how you fixed your issue.

**Comment by pakresi** [comment]
hello, i fixed, 
sd-webui api have bad documentation. 
i debug parameters from webui request and i wrote same to the api. 

its exacly same with my webui request and working good

```python
payload = {
  "init_images": [
    inputImageBase64
  ],
  "resize_mode": 0,
  "denoising_strength": 0.9,
  "image_cfg_scale": 1.5,
  "mask": maskImageBase64,
  "mask_blur": 0,
  "inpainting_fill": 3,
  "inpaint_full_res": 0,
  "inpaint_full_res_padding": 0,
  "inpainting_mask_invert": 0,
  "initial_noise_multiplier": 1.0,
  "prompt": prompt,
  "seed": -1,
  "subseed":-1,
  "subseed_strength":0,
  "seed_resize_from_h": 0,
  "seed_resize_from_w": 0,
  "sampler_name": "DPM++ 2M Karras",
  "batch_size": 1,
  "n_iter": 1,
  "steps": steps,
  "cfg_scale": 7,
  "width": int(orig_w),
  "height": int(orig_h),
  "negative_prompt": negative,
  "script_args": [],
  "sampler_index": 16,
  "include_init_images": True,
#  "script_name": "string",
  "send_images": True,
  "save_images": False,
      "alwayson_scripts": {
               "controlnet": {
                   "args": [ 
                     

                       {
                        "input_image": inputImageBase64,
                        "module": "openpose_full",
                        "model": "control_v11p_sd15_openpose [cab727d4]",
                        "weight": 1,
                        "lowvram": False,
                        "guidance_start": 0,
                        "guidance_end": 0.7,
                        "control_mode": "ControlNet is more important",
                        "resize_mode": 0,
                       "processor_res":512,
                         "pixel_perfect" : False
                        },
                     {
                        "input_image": segimageb64,
                        "module": "canny",
                        "model": "control_v11p_sd15_canny [d14c016b]",
                        "weight": 0.4,
                        "lowvram": False,
                        "threshold_a": 100,
                        "threshold_b": 200,
                        "guidance_start": 0.3,
                        "guidance_end": 0.6,
                        "control_mode": "My prompt is more important",
                        "resize_mode": 0,
                       "processor_res":512,
                         "pixel_perfect" : False
                      },
                       {
                        "input_image": inputImageBase64,
                        "module": "depth",
                        "model": "control_v11f1p_sd15_depth [cfd03158]",
                        "weight": 0.5,
                        "lowvram": False,
                        "guidance_start": 0.2,
                        "guidance_end": 0.6,
                        "control_mode": "My prompt is more important",
                        "resize_mode": 0,
                         "processor_res":512,
                         "pixel_perfect" : False
                      }
                    ]
                }
      }
  }
```

**Comment by sclausen** [comment]
What was your approach to debug the webui request? When I see the websocket messages, I just have a serialized data array with values I can't match to any keys 😅

**Comment by sclausen** [comment]
Thank you very much, you are a great help!

**Comment by h3clikejava** [comment]
> hello, i fixed, sd-webui api have bad documentation. i debug parameters from webui request and i wrote same to the api.
> 
> its exacly same with my webui request and working good
> 
> ```python
> payload = {
>   "init_images": [
>     inputImageBase64
>   ],
>   "resize_mode": 0,
>   "denoising_strength": 0.9,
>   "image_cfg_scale": 1.5,
>   "mask": maskImageBase64,
>   "mask_blur": 0,
>   "inpainting_fill": 3,
>   "inpaint_full_res": 0,
>   "inpaint_full_res_padding": 0,
>   "inpainting_mask_invert": 0,
>   "initial_noise_multiplier": 1.0,
>   "prompt": prompt,
>   "seed": -1,
>   "subseed":-1,
>   "subseed_strength":0,
>   "seed_resize_from_h": 0,
>   "seed_resize_from_w": 0,
>   "sampler_name": "DPM++ 2M Karras",
>   "batch_size": 1,
>   "n_iter": 1,
>   "steps": steps,
>   "cfg_scale": 7,
>   "width": int(orig_w),
>   "height": int(orig_h),
>   "negative_prompt": negative,
>   "script_args": [],
>   "sampler_index": 16,
>   "include_init_images": True,
> #  "script_name": "string",
>   "send_images": True,
>   "save_images": False,
>       "alwayson_scripts": {
>                "controlnet": {
>                    "args": [ 
>                      
> 
>                        {
>                         "input_image": inputImageBase64,
>                         "module": "openpose_full",
>                         "model": "control_v11p_sd15_openpose [cab727d4]",
>                         "weight": 1,
>                         "lowvram": False,
>                         "guidance_start": 0,
>                         "guidance_end": 0.7,
>                         "control_mode": "ControlNet is more important",
>                         "resize_mode": 0,
>                        "processor_res":512,
>                          "pixel_perfect" : False
>                         },
>                      {
>                         "input_image": inputImageBase64,
>                         "module": "canny",
>                         "model": "control_v11p_sd15_canny [d14c016b]",
>                         "weight": 0.4,
>                         "lowvram": False,
>                         "threshold_a": 100,
>                         "threshold_b": 200,
>                         "guidance_start": 0.3,
>                         "guidance_end": 0.6,
>                         "control_mode": "My prompt is more important",
>                         "resize_mode": 0,
>                        "processor_res":512,
>                          "pixel_perfect" : False
>                       },
>                        {
>                         "input_image": inputImageBase64,
>                         "module": "depth",
>                         "model": "control_v11f1p_sd15_depth [cfd03158]",
>                         "weight": 0.5,
>                         "lowvram": False,
>                         "guidance_start": 0.2,
>                         "guidance_end": 0.6,
>                         "control_mode": "My prompt is more important",
>                         "resize_mode": 0,
>                          "processor_res":512,
>                          "pixel_perfect" : False
>                       }
>                     ]
>                 }
>       }
>   }
> ```
> 
> you can change controlnet parameters

You are the god in my heart.
